### DISCUSSION TITLE
Create Table with dynamic column list

### DISCUSSION THREAD

**Discussion by StefanWoe** [fnc12/sqlite_orm#1003] [Q&A]
Hello,

i have a vector with information describing the columns. How can i use this to create a table?

All examples i see have hard coded columns which are passed as tuples. By my knowledge tuples are compile time created.

Is that possible?

Best regards
Stefan

**Comment by fnc12** [comment]
No it is not possible in `sqlite_orm`. Actually in all cases we met here when people needed dynamic schema it turned out that it could be solved using static schema. So you can try to provide your task and we can think about it

**Reply by rzahraie** [reply]
Hello

First, thank you for creating this excellent framework.

To piggyback on StefanWoe's question, one scenario for dynamic table or column creation is reflection. Specifically querying Google's Protobuf messages or objects at runtime and creating columns based on the obtained information. I'm not sure how a static schema can be used in such a scenario.

Please advise.

Thanks again.

**Reply by fnc12** [reply]
google protobuf has static schema. I mean its' fields are defined at compilation time so I don't see any reason of using dynamic scheme as feature for your case. Please share some code if I understood you wrong

**Reply by rzahraie** [reply]
Thank you for the prompt reply.

Yes, Protobuf has static schemas. However we would like to automate the generation of the database tables as much as possible. It is conceivable we may have a large number of .proto files. If so, then the creation of their corresponding tables through explicit declaration can be quite tedious and time consuming.

It is also possible that Protobuf messages will be received by the application which their .proto files will not be available at compile time. Discovery of these messages can only be achieved through reflection at runtime.

Can some type of a macro be helpful in such situations?

**Reply by fnc12** [reply]
one can make a `.proto` parser to generate C++ code with `sqlite_orm`. But one can not make dynamic schema cause dynamic schema will lost all static features like type validation
